ansible/roles/sysprep/tasks/main.yml

```diff
@@ -42,6 +42,19 @@
       state: touch
       mode: "{{ machine_id_mode }}"
 
+- name: Truncate D-Bus machine-id
+  file:
+    path: /var/lib/dbus/machine-id
+    state: absent
+  when: ansible_os_family == "Debian"
+
+- name: Link D-Bus machine-id to /etc/machine-id
```

Comment:
Please add a comment explaining why we need to do this.

Reply:
Is this only needed for Ubuntu on VMWare?

Reply:
See above comment. I believe this may be required for _all_ Ubuntu images to ensure consistency. 

In some Ubuntu 20.04 systems in my homelab, this symlink seems to be already present but the common denominator is that all these were created from the "desktop" ISO. Reading a comment on an old Launchpad bug (https://bugs.launchpad.net/ubuntu/+source/dbus/+bug/1508697/comments/4) would explain this:

>  * /usr/lib/tmpfiles.d/dbus.conf will create the /var/lib/dbus/machine-id symlink to /etc/machine-id. I. e. this would happen on "first boot" *if* /var/lib/dbus/machine-id does not exist yet. This is what happens if you install from a desktop CD: livecd-rootfs removes machine-id from the image, so the first boot goes through that tmpfiles.d snippet and creates the symlink.
> 
>  * If you install server, that actually apt-get installs stuff including dbus. dbus.postinst calls "dbus-uuidgen --ensure" which according to NEWS
> 
>      • in "dbus-uuidgen --ensure", try to copy systemd's /etc/machine-id
>        to /var/lib/dbus/machine-id instead of generating an entirely new ID
>        (fd.o #77941, Simon McVittie)

Which explains why on a stock server install, the symlink is not present and it is a regular file with the same content as /etc/machine-id (https://www.freedesktop.org/software/systemd/man/systemd-machine-id-setup.html explains how /etc/machine-id can be recreated from the DBUS machine-id if it is not also cleaned.)

My thinking on this is that as RHEL seems to do the symlink anyway, and some Ubuntu installs also will do this depending on the image used, it's good to ensure it's done in a consistent way across all variations of Ubuntu images.